### PR TITLE
Fixes #RHIROS-833 - upload coverage to codecov in github workflow

### PR DESCRIPTION
**Pull request by kgaikwad**
Referred - https://github.com/codecov/codecov-action

### CODE REVIEW

**Comment by kgaikwad** [comment]
+ adding @digitronik in cc as he requested this change. 
@digitronik, could you also confirm if ros-backend populated into projects list which enabled codecov and % value? :)

**Review by digitronik — APPROVED** [review]
Thanks for this PR
I can see code coverage report https://app.codecov.io/github/RedHatInsights/ros-backend/pulls

**Comment by saltgen** [comment]
@kgaikwad Coverage is at 72.51%, we are almost there, 75-80 is a good number :raised_hands:

@digitronik Do we have the below warning cause this is the PR that enables coverage and there is currently no base?
`! No coverage uploaded for pull request base (main@d234b40)`

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -80,3 +80,8 @@ jobs:
         DB_PORT: 15432
         DB_USER: postgres
         DB_PASSWORD: postgres
+    - name: Upload coverage to codecov
+      uses: codecov/codecov-action@v3
+      with:
+        fail_ci_if_error: true
+        verbose: true
```
Comment (saltgen) [inline .github/workflows/build.yml:87]:
```suggestion
        verbose: true
        flags: unittests
```

Comment (saltgen) [reply .github/workflows/build.yml:87]: @kgaikwad Although this is an optional suggestion, IMO it helps provide a legible context/metadata for the coverage data

**Comment by kgaikwad** [comment]
> @digitronik Do we have the below warning cause this is the PR that enables coverage upload and there is currently no base?
> `! No coverage uploaded for pull request base (main@d234b40)`

@saltgen  - 
This warning shown because the base commit i.e main@d234b40 did not upload coverage. 
As per their documentation, it is mentioned that in order to produce an accurate coverage comparison, Codecov must have the coverage report for the pull request base.  I hope this information clarifies your query.

**Comment by digitronik** [comment]
exactly,  We don't have a reference for comparison.

**Comment by kgaikwad** [comment]
@saltgen - Do you have any additional comments?

**Comment by saltgen** [comment]
I'm good, thanks @kgaikwad @digitronik

**Review by saltgen — APPROVED** [review]

**Pull request merged**